libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -159,25 +189,75 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
     options: SourceOptions
   ): SetConfigurationSettingParam<FeatureFlagValue>[] {
     const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+    const featureFlags = new Array<FeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.foundMsFeatureManagement) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaSection(config);
+
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} is in an invalid format: id is a required property`
+            );
+          }
+
+          if (featureFlags.some(existingFeatureFlag => existingFeatureFlag.id === featureFlag.id)) {
+            throw new ArgumentError(`Duplicate feature flag with id '${featureFlag.id}' is found.`);
+          }
+
+          if (!this.validateFeatureName(featureFlag.id)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature id.`
+            );
+          }
+          featureFlags.push(this.getFeatureFlagDefinitionFromMsFmSchema(featureFlag));
+        }
       }
-      const prefix: string = options.prefix ?? "";
-      settings.push({
-        key: featureFlagPrefix + prefix + featureFlag,
+    }
+
+    if (this.legacySchemaFeatureManagementKeyWord) {
+      const legacySchemaFeatureFlags = this.getLegacySchema(config);
+
+      if (legacySchemaFeatureFlags) {
+        for (const featureFlag in legacySchemaFeatureFlags) {
+          if (!this.validateFeatureName(featureFlag)) {
+            throw new ArgumentError(
+              `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
```

Comment:
Feature flag id contains invalid character,'%' and ':' are not allowed

Reply:
Should we not add the id found in the error message?